compiler/rustc_hir_typeck/src/writeback.rs

```diff
@@ -881,6 +881,8 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for EagerlyNormalizeConsts<'tcx> {
     }
 
     fn fold_const(&mut self, ct: ty::Const<'tcx>) -> ty::Const<'tcx> {
-        self.tcx.try_normalize_erasing_regions(self.param_env, ct).unwrap_or(ct)
+        self.tcx
+            .try_normalize_erasing_regions(ty::TypingEnv::from_param_env(self.param_env), ct)
```

Comment:
should we pass in the typing env of the infcx we're currently writeback-ing?

Reply:
I don't think so, we're defining opaque types here, so using a typing env instead of a canonical query just can't work. I treat all `TypingEnv::from_param_env` as implicit fixmes.